src/gpio/mod.rs

```diff
@@ -171,91 +170,142 @@ macro_rules! impl_pin {
+                pupdr: &mut PUPDR<$GPIOX>,
+            ) -> $PXi<Input<PullUp>> {
                 Self::set_input_mode(moder);
                 // pull-up
-                pupdr.pupdr().modify(|r, w| unsafe { w.bits((r.bits() & !(0b11 << Self::OFFSET)) | (0b01 << Self::OFFSET)) });
+                pupdr.pupdr().modify(|r, w| unsafe {
+                    w.bits((r.bits() & !(0b11 << Self::OFFSET)) | (0b01 << Self::OFFSET))
+                });
 
                 $PXi(PhantomData)
             }
 
             #[inline]
             fn set_output_mode(moder: &mut MODER<$GPIOX>) {
-                moder.moder().modify(|r, w| unsafe { w.bits((r.bits() & !(0b11 << Self::OFFSET)) | (0b01 << Self::OFFSET)) });
+                moder.moder().modify(|r, w| unsafe {
+                    w.bits((r.bits() & !(0b11 << Self::OFFSET)) | (0b01 << Self::OFFSET))
+                });
             }
 
             /// Configures the pin to operate as an open drain output pin
-            pub fn into_open_drain_output(self, moder: &mut MODER<$GPIOX>, otyper: &mut OTYPER<$GPIOX>) -> $PXi<Output<OpenDrain>> {
+            pub fn into_open_drain_output(
+                self,
+                moder: &mut MODER<$GPIOX>,
+                otyper: &mut OTYPER<$GPIOX>,
+            ) -> $PXi<Output<OpenDrain>> {
                 Self::set_output_mode(moder);
                 // open drain output
-                otyper.otyper().modify(|r, w| unsafe { w.bits(r.bits() | (0b1 << $i)) });
+                otyper
+                    .otyper()
+                    .modify(|r, w| unsafe { w.bits(r.bits() | (0b1 << $i)) });
 
                 $PXi(PhantomData)
             }
 
             /// Configures the pin to operate as an push pull output pin
-            pub fn into_push_pull_output(self, moder: &mut MODER<$GPIOX>, otyper: &mut OTYPER<$GPIOX>) -> $PXi<Output<PushPull>> {
+            pub fn into_push_pull_output(
+                self,
+                moder: &mut MODER<$GPIOX>,
+                otyper: &mut OTYPER<$GPIOX>,
+            ) -> $PXi<Output<PushPull>> {
                 Self::set_output_mode(moder);
                 // push pull output
-                otyper.otyper().modify(|r, w| unsafe { w.bits(r.bits() & !(0b1 << $i)) });
+                otyper
+                    .otyper()
+                    .modify(|r, w| unsafe { w.bits(r.bits() & !(0b1 << $i)) });
 
                 $PXi(PhantomData)
             }
 
             // alternate function mode
             fn set_alt_fun_mode(moder: &mut MODER<$GPIOX>, afr: &mut $AFR<$GPIOX>, af: u32) {
-                moder.moder().modify(|r, w| unsafe { w.bits((r.bits() & !(0b11 << Self::OFFSET)) | (0b10 << Self::OFFSET)) });
-                afr.afr().modify(|r, w| unsafe { w.bits((r.bits() & !(0b1111 << Self::OFFSET)) | (af << Self::OFFSET)) });
+                moder.moder().modify(|r, w| unsafe {
+                    w.bits((r.bits() & !(0b11 << Self::OFFSET)) | (0b10 << Self::OFFSET))
```

Comment:
I remember you mentioned error in alternative function mode, will you make a separate PR for that?